Azure/CanadaPubSecALZ
# Diagnostic Settings Policies for Azure App Service & Function App

## Overview/Summary

- Azure Policy definitions to enable diagnostic settings for logs through automation.  There are two policy definitions added in this PR:

  - Policy for App Service
  - Policy for Function App 

- Update existing audit diagnostic settings policy to monitor for `Microsoft.Web/sites` resource type.

## This PR fixes/adds/changes/removes

Fixes #132

### Breaking Changes

None

## Testing Evidence

1. Create Azure Policy definitions for App Service & Function App

![image](https://user-images.githubusercontent.com/16688027/149026009-27dd66d0-1a8a-4bfa-849f-70776799394d.png)

2. Update Log Analytics custom initiative to include 2 new policies

![image](https://user-images.githubusercontent.com/16688027/149026107-30bb1a95-97eb-4a9e-85db-d07e55d48eb0.png)

3. Verify that policy validates resources that are missing diagnostic settings

![image](https://user-images.githubusercontent.com/16688027/149026337-041fb4c3-d5d8-4ef1-9aed-901b1886b2e7.png)

![image](https://user-images.githubusercontent.com/16688027/149026389-1188d15a-8a92-43ab-bd24-c2d6f47c17a5.png)

![image](https://user-images.githubusercontent.com/16688027/149026421-4d942dbc-d680-47ff-b827-547cb7d734fe.png)


4.  Verify that policy remediates missing diagnostic settings

![image](https://user-images.githubusercontent.com/16688027/149026755-23844434-edf5-4f7d-95a6-c553f20898a0.png)

![image](https://user-images.githubusercontent.com/16688027/149026840-74c1b396-7c76-4cce-9389-3acedbefc8fb.png)

![image](https://user-images.githubusercontent.com/16688027/149027031-2393c41a-932a-4138-b642-ace8b0dd0cdf.png)

![image](https://user-images.githubusercontent.com/16688027/149026883-6996286a-dd7a-4ad3-94fd-09ba4a314bef.png)

![image](https://user-images.githubusercontent.com/16688027/149026982-e77c913c-4464-4b69-9599-66ab4f1a7a8e.png)


## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/azure/CanadaPubSecALZ/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/azure/CanadaPubSecALZ/issues)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/azure/CanadaPubSecALZ/tree/main)
